wordpress-mobile/WordPress-Android
# Adds FluxC as composite build

This PR adds FluxC as a possible composite build. This is meant to make local development easier with the following main benefits:
* Being able to make changes to the FluxC codebase from the same Android Studio instance that is running WPAndroid
* Making changes to FluxC repo and immediately being able to use it within WPAndroid
* Being able to use IDE goodies such as "Find usages"

_*Edit:* The following sections are modified after the changes in 60f9b583641bf5722bb06676f9644058a4436fd2 & d7498355756702970c093e931b989dc50f101adb which simplified the process._

This is accomplished by:
* Including the FluxC project by its relative path by adding it to `local-builds.gradle` file which will make it available for use within the same Android Studio instance

Please see the notes in `local-builds.gradle-example` for more information.

**To test:**
* Copy `local-builds.gradle-example` as `local-builds.gradle` and update `localFluxCPath` to your liking
* Use `Navigate Class` shortcut to open `PostStore` which is in FluxC and add a new method such as a summation
* Use `Navigate Class` shortcut to open `PostListViewModel` which is in WPAndroid and use that function
* Run the project and verify that it works 🎉 
* Comment out `localFluxCPath` in `local-builds.gradle` file and verify that the build fails for missing the new method from `PostStore`
* Discard the changes for `PostListViewModel`, build the app and verify that it works

_P.S: Unfortunately making changes to FluxC is _very_ slow to compile. We can improve the builds and this won't be as big of an issue, but this probably won't happen at least until the end of year._

**PR submission checklist:**
